tests/e2e/mapper/mapper_test.go

```diff
@@ -0,0 +1,115 @@
+package mapper
+
+import (
+	"bytes"
+	"encoding/json"
+	"fmt"
+	"io/ioutil"
+	"net/http"
+	"path"
+	"reflect"
+	"time"
+
+	MQTT "github.com/eclipse/paho.mqtt.golang"
+	. "github.com/onsi/ginkgo"
+	. "github.com/onsi/gomega"
+
+	"github.com/kubeedge/kubeedge/edge/test/integration/utils/common"
+	"github.com/kubeedge/kubeedge/edge/test/integration/utils/helpers"
+	"github.com/kubeedge/kubeedge/tests/e2e/utils"
+)
+
+var TokenClient Token
+var ClientOpts *MQTT.ClientOptions
+var Client MQTT.Client
+var timesSpecified int
+var timesExecuted int
+
+type ScheduleResult struct {
+	EventName   string
+	TimeStamp   int64
+	EventResult string
+}
+
+var scheduleresult ScheduleResult
+var msgreceived bool
+
+func DataReceived(client MQTT.Client, message MQTT.Message) {
+	topic := "$ke/device/bluetooth-mapper/sensortagmock/scheduler/result"
+	expectedTemp := "36"
+	if message.Topic() == topic {
+		msgreceived = false
+		timesExecuted++
+		devicePayload := message.Payload()
+		err := json.Unmarshal(devicePayload, &scheduleresult)
+		if err != nil {
+			common.Failf("Unmarshall failed %s", err)
+		} else {
+			if reflect.DeepEqual(scheduleresult.EventResult, expectedTemp) {
+				msgreceived = true
+			}
+		}
+	}
+}
+
+var _ = Describe("Application deployment test in E2E scenario", func() {
+
+	Context("Test read characteristic of mapper", func() {
+		BeforeEach(func() {
+			ClientOpts = helpers.HubClientInit(ctx.Cfg.MqttEndpoint, "bluetoothmapper", "", "")
+			Client = MQTT.NewClient(ClientOpts)
+			if TokenClient = Client.Connect(); TokenClient.Wait() && TokenClient.Error() != nil {
+				common.Failf("client.Connect() Error is %s", TokenClient.Error())
+			} else {
+				fmt.Println("Connection successful")
+			}
+			Expect(TokenClient.Error()).NotTo(HaveOccurred())
+			scheduletopic := "$ke/device/bluetooth-mapper/sensortagmock/scheduler/result"
+			Token := Client.Subscribe(scheduletopic, 0, DataReceived)
+			time.Sleep(1 * time.Minute)
```

Comment:
can we reduce the sleep here ? 1 minute is too long.

Reply:
Updated

Reply:
Updated